Makefile

```diff
@@ -0,0 +1,83 @@
+clean: clean-build clean-py clean-test
+
+clean-build:
+	rm -fr build/
+	rm -fr dist/
+	rm -fr .eggs/
+	find . -name '*.egg-info' -exec rm -fr {} +
+	find . -name '*.egg' -exec rm -f {} +
+
+clean-py:
+	find . -name '*.pyc' -exec rm -f {} +
+	find . -name '*.pyo' -exec rm -f {} +
+	find . -name '__pycache__' -exec rm -fr {} +	
+
+clean-test:
+	find . -name '.pytest_cache' -exec rm -fr {} +
+	find . -name '.coverage' -exec rm -fr {} +
+	find . -name 'htmlcov' -exec rm -fr {} +
+	find . -name '.cache' -exec rm -fr {} +
+
+format:
```

Comment:
These require a venv to run, shold we have some sort of check to make sure it exists before running the rest of this command + others below?